lib/internal/encoding.js

```diff
@@ -69,7 +70,49 @@ const CONVERTER_FLAGS_FLUSH = 0x1;
 const CONVERTER_FLAGS_FATAL = 0x2;
 const CONVERTER_FLAGS_IGNORE_BOM = 0x4;
 
-const empty = new Uint8Array(0);
+const empty = new FastBuffer();
+
+// Has to be synced with src/
+const namesSinglebyte = [
+  'ibm866',
+  'koi8-r',
+  'koi8-u',
+  'macintosh',
+  'x-mac-cyrillic',
+  'iso-8859-2',
+  'iso-8859-3',
+  'iso-8859-4',
+  'iso-8859-5',
+  'iso-8859-6',
+  'iso-8859-7',
+  'iso-8859-8',
+  'iso-8859-8-i',
+  'iso-8859-10',
+  'iso-8859-13',
+  'iso-8859-14',
+  'iso-8859-15',
+  'iso-8859-16',
+  'windows-874',
+  'windows-1250',
+  'windows-1251',
+  'windows-1252',
+  'windows-1253',
+  'windows-1254',
+  'windows-1255',
+  'windows-1256',
+  'windows-1257',
+  'windows-1258',
+  'x-user-defined', // Has to be last, special case
+];
```

Comment:
I guess you don’t need this array long term. Can you create it inline with the Set?